### DISCUSSION TITLE
Consider doing a shallow clone on BuildKite CI

### DISCUSSION THREAD

**Discussion by bryanhonof** [MinaProtocol/mina#11739] [Ideas]
Currently, I think at least, the BuildKite CI does a full checkout of the git repo. We could [add a flag to the BuildKite agents](https://buildkite.com/docs/agent/v3/configuration#git-clone-flags) that do a shallow clone of depth 1, so we only get the blobs in there corresponding to the commit we are doing to execute the CI on. Would that make sense?

**Comment by mrmr1993** [comment]
I suspect that this will break the 'mergeability' checks on CI, which need to have the history and the `compatible` and `develop` branches available.

**Reply by bryanhonof** [reply]
That makes sense. But that's the only check with that requirement, right? Perhaps we could enable it for every other job, except that one?
